packages/common/src/index.ts

```diff
@@ -294,3 +294,17 @@ export enum VerifiedCredentialState {
   INVALID,
   PENDING,
 }
+
+export const convertStatusToText = (applicationStatus: string | number) => {
```

Comment:
`convertStatusToText` does not coerce a string into a number for the switch statement. If applicationStatus is a string, the switch statement will not match any cases, and the function will return "PENDING" due to the default case.

Perhaps we should use an enum here for application status. Maybe something like:
```
enum ApplicationStatus {
  PENDING = "0",
  APPROVED = "1",
  REJECTED = "2",
  CANCELLED = "3",
}

export const convertStatusToText = (applicationStatus: string | number): string => {
  // Ensure the applicationStatus is a string
  applicationStatus = applicationStatus.toString();

  switch (applicationStatus) {
    case ApplicationStatus.CANCELLED:
      return "CANCELLED";
    case ApplicationStatus.REJECTED:
      return "REJECTED";
    case ApplicationStatus.APPROVED:
      return "APPROVED";
    case ApplicationStatus.PENDING:
    default:
      return "PENDING";
  }
};
```